src/connection_string.ts

```diff
@@ -590,10 +591,24 @@ export const OPTIONS = {
   serverApi: {
     target: 'serverApi',
     transform({ values: [version] }): ServerApi {
-      if (typeof version === 'string') {
-        return { version };
+      const serverApiToValidate =
+        typeof version === 'string' ? ({ version } as ServerApi) : (version as ServerApi);
+      const versionToValidate = serverApiToValidate && serverApiToValidate.version;
+      if (!versionToValidate) {
+        throw new MongoParseError(
+          `Invalid \`serverApi\` property; must specify a version as one of the following strings: "${Object.values(
+            ServerApiVersion
+          ).join('", "')}"`
+        );
+      }
+      if (!Object.values(ServerApiVersion).some(v => v === versionToValidate)) {
+        throw new MongoParseError(
+          `Invalid server API version=${versionToValidate}; must be one of the following strings: "${Object.values(
```

Comment:
Just a suggestion, maybe we could tighten up the error message here a bit:
```
Invalid server API version=${versionToValidate}; must be of the following: "a", "b"
```
It avoids the issue with plural and I think the quotes imply the type must be string, along with our documentation saying as much.